Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -419,6 +419,55 @@ public let CodeGenerators: [CodeGenerator] = [
         b.destruct(arr, selecting: indices, into: candidates, hasRestElement: probability(0.2))
     },
 
+    CodeGenerator("DestructObjectGenerator", input: .object()) { b, obj in
+        var properties = Set<String>()
+        for _ in 0...5 {
+            if let prop = b.type(of: obj).properties.randomElement(), !properties.contains(prop){
```

Comment:
Nit: I don't think you need the `!properties.contains(prop)` condition, `properties.insert(prop)` will already need to do that check, and be a nop if the element already exists

Reply:
Alternatively, you could perform `properties.insert(b.genPropertyNameForRead())` when the property already exists or if none was found? Maybe that'd be a better option.

Reply:
yes, you're right it is a Set :)

I'm a little hesitant to add random properties using genPropertyName*, mainly because I feel it might cause a high failure rate if the property does not exist, especially when running this generator on imported compiled programs. I think we might need a better way to track properties across objects/classes so that assignments are more accurate, wdyt?

Reply:
So I think it's fine (but maybe only if `!conservative_mode`). Due to mutations, e.g. InputMutator, we'll anyway quickly (?) end up with different objects being destructured or "wrong" property names being used. But that's "fine", it'll just result in a variable with the `undefined` value afaik. On the other hand, I'm a bit worried about only limiting us to what the AI knows about object properties. Improvements to the AI will still automatically improve this logic later on (because we'll only use `b.genPropertyNameForRead()` as fallback).

Reply:
Fair, that makes sense to me